https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=172034


[ backported ]
revision 1.83
date: 2006/07/15 16:33:24;  author: hjl;  state: Exp;  lines: +34 -10
2006-07-15  H.J. Lu  <hongjiu.lu@intel.com>

	PR binutils/2829
	* i386-dis.c (GRP11_C6): NEW.
	(GRP11_C7): Likewise.
	(GRP12): Updated.
	(GRP13): Likewise.
	(GRP14): Likewise.
	(GRP15): Likewise.
	(GRP16): Likewise.
	(GRPAMD): Likewise.
	(GRPPADLCK1): Likewise.
	(GRPPADLCK2): Likewise.
	(dis386): Use GRP11_C6 and GRP11_C7 for entres 0xc6 and 0xc7,
	respectively.
	(grps): Add entries for GRP11_C6 and GRP11_C7.


diff -rup gdb-6.5.orig/opcodes/i386-dis.c gdb-6.5/opcodes/i386-dis.c
--- gdb-6.5.orig/opcodes/i386-dis.c	2006-09-28 15:50:43.000000000 -0400
+++ gdb-6.5/opcodes/i386-dis.c	2006-09-28 15:56:00.000000000 -0400
@@ -420,12 +420,14 @@ fetch_data (struct disassemble_info *inf
 #define GRP9	  NULL, NULL, USE_GROUPS, NULL, 16, NULL, 0
 #define GRP10	  NULL, NULL, USE_GROUPS, NULL, 17, NULL, 0
 #define GRP11	  NULL, NULL, USE_GROUPS, NULL, 18, NULL, 0
-#define GRP12	  NULL, NULL, USE_GROUPS, NULL, 19, NULL, 0
-#define GRP13	  NULL, NULL, USE_GROUPS, NULL, 20, NULL, 0
-#define GRP14	  NULL, NULL, USE_GROUPS, NULL, 21, NULL, 0
-#define GRPAMD	  NULL, NULL, USE_GROUPS, NULL, 22, NULL, 0
-#define GRPPADLCK1 NULL, NULL, USE_GROUPS, NULL, 23, NULL, 0
-#define GRPPADLCK2 NULL, NULL, USE_GROUPS, NULL, 24, NULL, 0
+#define GRP11_C6  NULL, NULL, USE_GROUPS, NULL, 19, NULL, 0
+#define GRP11_C7  NULL, NULL, USE_GROUPS, NULL, 20, NULL, 0
+#define GRP12	  NULL, NULL, USE_GROUPS, NULL, 21, NULL, 0
+#define GRP13	  NULL, NULL, USE_GROUPS, NULL, 22, NULL, 0
+#define GRP14	  NULL, NULL, USE_GROUPS, NULL, 23, NULL, 0
+#define GRPAMD	  NULL, NULL, USE_GROUPS, NULL, 24, NULL, 0
+#define GRPPADLCK1 NULL, NULL, USE_GROUPS, NULL, 25, NULL, 0
+#define GRPPADLCK2 NULL, NULL, USE_GROUPS, NULL, 26, NULL, 0
 
 #define PREGRP0   NULL, NULL, USE_PREFIX_USER_TABLE, NULL,  0, NULL, 0
 #define PREGRP1   NULL, NULL, USE_PREFIX_USER_TABLE, NULL,  1, NULL, 0
@@ -739,8 +741,8 @@ static const struct dis386 dis386[] = {
   { "retT",		XX, XX, XX },
   { "les{S|}",		Gv, Mp, XX },
   { "ldsS",		Gv, Mp, XX },
-  { "movA",		Eb, Ib, XX },
-  { "movQ",		Ev, Iv, XX },
+  { GRP11_C6 },
+  { GRP11_C7 },
   /* c8 */
   { "enterT",		Iw, Ib, XX },
   { "leaveT",		XX, XX, XX },
@@ -1453,6 +1455,28 @@ static const struct dis386 grps[][8] = {
     { "pslld",	MS, Ib, XX },
     { "(bad)",	XX, XX, XX },
   },
+  /* GRP11_C6 */
+  {
+    { "movA",	Eb, Ib, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+  },
+  /* GRP11_C7 */
+  {
+    { "movQ",	Ev, Iv, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+    { "(bad)",	XX, XX, XX },
+  },
   /* GRP12 */
   {
     { "(bad)",	XX, XX, XX },
